drivers/i2c/i2c_sam0.c

```diff
@@ -137,6 +139,17 @@ static void i2c_sam0_isr(const struct device *dev)
 		return;
 	}
 
+	/*
+	 * Directly send/receive next message if it is in the same direction and
+	 * the current message has no stop or restart flags.
+	 */
+	const bool continue_next = (data->msg.size == 1) && (data->num_msgs > 1) &&
+				   ((data->msgs[0].flags & I2C_MSG_RW_MASK) ==
+				    (data->msgs[1].flags & I2C_MSG_RW_MASK)) &&
+				   !(data->msgs->flags & I2C_MSG_STOP) &&
+				   !(data->msgs->flags & I2C_MSG_RESTART) &&
```

Comment:
The restart flag should be checked on the next message, not the current one.

Reply:
Fixed, also updated line 149 to make it more readable.